### ISSUE TITLE
pre-training for data params

### ISSUE THREAD

**Issue by KisaragiDev** [kisaragiRY/decoding-DG#46]
# Unsupervised learning for choosing data parameters 
* **Method:** K-means clustering
* **Score:** model's inertia (Sum of squared distances of samples to their closest cluster center)
* **Parameters:** 
    * `window size`: window size of Gaussian kernel for smoothing data 
    * `K`: the threshold for the length of segments
    * `num_kernels`: number of kernels used in ROCKET transformation
* **Criteria:** the smaller the inertia of the clustering model, the better.
* **working dir:** `/work/notebooks/3-decoding-model/2-classification/3-time-series-classification/2-pre-training`

## Work Flow
- [x] research on how to use K-means 
- [x] train script for testing on diff params of data -> `train.py`
   1. large scale of num_kernels and K 
   2. ~small scale~
- [x] Analyze the training results -> `02-influence-of-data-param.ipynb`
   - visualized the results and it's the opposite of what we expected
   1. ~get the best param pair from the large scale~
   2. ~examine the small scale results~
- [ ] ~Analyze the similarity between the clustering results and y(position) -> `03-similarity-between-clusters-positions.ipynb`~
   * ~coverage rate, e.g probability of instances from the same position are in the same cluster~
   * ~if the coverage rate is above 50% ?~

**Comment by KisaragiDev** [comment]
# K-means usage
```python
from sklearn.cluster import KMeans

kmeans = KMeans(n_clusters=2, random_state=0, n_init="auto")
kmeans.fit(X_train)

# criterion
kmeans.inertia_
```

## Regarding Data
1. **Use all the data to train a K-means model** -> Modify `UniformSegmentDataset`
   - [x]  change `train_ratio` to Optional
   - [x]  if it's given a float number then split the data
   - [x]  if not, then only return the train set which used all the available data.
   - [x] import the modified `UniformSegmentDataset`

2. **The `/data` folder was deleted**
    - [x] convert the original xlsx data to csv 
    :point_right: It takes Wall time of **266 ms** to read from csv files of all mice, **45.4 s**for xlsx files

**Comment by KisaragiDev** [comment]
# K-means training
1. the cv was not used in the training
   However, If we are to use CV for K-means: [k-fold Cross Validation for determining k in k-means?](https://stackoverflow.com/questions/6629165/k-fold-cross-validation-for-determining-k-in-k-means)
    1. A ground truth set is needed
    2. Specify a scoring method.

## checking the results
**The desired result:** the inertia would reduce as more kernels are used
Reasons
* the small the inertia is, the data is more likely to be separated
* the previous supervised method showed the higher the num_kernels is, the more accurate the model is.
👉  **However** the opposite result was shown

## Questions to be answered
1.  is it true that `the small the inertia is, the data is more likely to be separated`?
2. how did `num_kernels` influenced inertia?

### Answers(some ideas)
1. Smaller inertia means the samples can be separated geometrically more easily. 
   - However, relationship between the samples in this research may **not be geometrical**
   - therefore, even the inertia is small when less kernels were used, it doesn't necessarily lead to high accuracy.
2. unknown
   - But it's clearly there is some relations between the `num_kernels` and inertia 
   - the more kernels used, the samples are distributed more scatteredly.

**Comment by KisaragiDev** [comment]
The n_clusters parameter was set to a wrong number, retrained the k-means model after modifying.
